crates/runtime/src/dataconnector.rs

```diff
@@ -456,6 +459,19 @@ pub trait DataConnector: Send + Sync {
     ) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
         Ok(())
     }
+
+    /// Returns a `MetricsProvider` for the data connector.
+    ///
+    /// If the data connector does not support metrics, return `None`.
+    fn metrics_provider(&self) -> Option<Arc<dyn MetricsProvider>> {
+        None
+    }
+}
+
+impl<T: DataConnector + Debug + 'static> MetricsProviderComponent for T {
+    fn metrics_provider(&self) -> Option<Arc<dyn MetricsProvider>> {
+        self.metrics_provider()
+    }
 }
```

Comment:
This implementation and the `MetricsProviderComponent` isn't technically required for this PR - but if would be needed to go further down the path of a "Component" trait.